### ISSUE TITLE
Add Non-symmetrical Plant Faces

### ISSUE THREAD

**Issue by ghost** [minetest/minetest#13040]
##### Minetest version
```
Minetest 5.6.1
```

##### Summary
When you look closely at any `plantlike` node, they look symmetrical when looked at from certain angles.

This is how a `plantlike` node looks like from a normal angle.

![Normal Angle](https://user-images.githubusercontent.com/114245896/206689296-24b8607c-f662-40c4-a538-b0ec8b9166fd.png)

This is how it looks like from one of the weird angles.

![Symmetrical 1](https://user-images.githubusercontent.com/114245896/206689456-7c4cc6e3-dff7-4354-ba99-e9b432c85151.png)

This is how it looks like on the opposite side of the weird angle.

![Symmetrical 2](https://user-images.githubusercontent.com/114245896/206689532-83e5023d-1a09-4da4-899e-d08a6fa7052b.png)

**Comment by fluxionary** [comment]
This is basically how it's intended to look. You've got two 2d textures overlapping in an X pattern. 

If you want your plants to not appear like this, you can give them the paramtype2 "meshoption": 

https://github.com/minetest/minetest/blob/504e43e0dae50ad98e19db0649c9c825cf2ab7a7/doc/lua_api.txt#L1123-L1141

**Comment by fluxionary** [comment]
> That does not appear to solve the problem that I'm having.

true, but i hoped perhaps something there might help you get something that looked better to you.

i guess what you're asking for, is that the backfaces in the plantlike drawtype to not be "flipped".

**Comment by TurkeyMcMac** [comment]
Could be fixed by allowing two different tiles for plantlike nodes?

**Comment by Desour** [comment]
> What I was thinking of, was something among the lines of this: https://www.reddit.com/r/Minecraft/comments/wvl4cx/so_i_was_making_a_pathway_when_i_noticed/

Note that using flipped / non-mirrored uv coords on the backface has the disadvantages of looking weird and making shadows of only one of the sides.

**Comment by fluxionary** [comment]
> Note that using flipped / non-mirrored uv coords on the backface has the disadvantages of looking weird and making shadows of only one of the sides.

very good point. but we have to remember that the 3d world in minetest is a series of approximations, and that none of those approximations will match all the assumptions we make about how things approximate IRL 3D.

personally, i don't have an intuition on what is "most correct" here, but the issue is interesting and describable.

**Comment by Desour** [comment]
> What does this mean? I don't really understand all the jargon.

> > using flipped / non-mirrored uv coords on the backface

= what was seen in the video

(UV-coordinates are 2d positions (specified at each corner, and then mixed at each rendered pixel) to map the rendered pixels to positions in the plant image. With backface I meant here "the other side".)

>>has the disadvantages of looking weird

= You look at a blossom pixel from one side and on the other side it's not there.

>>and making shadows of only one of the sides.

= The shape of the plant's shadow is the shape of the image on the shadowy side of the plant. This is inconsistent if you look from the bright side.

**Comment by runsy** [comment]
There are tricks to make it look better. In the case of the plant in your example avoid the leaves in the 2 vertical center lines.

https://ih1.redbubble.net/image.896266789.0377/flat,750x,075,f-pad,750x1000,f8f8f8.jpg

In the case of the flowers, which you could intend to make quite symmetrical in principle, make them one pixel thicker, or asymmetrical.

https://ih1.redbubble.net/image.1122948676.2695/flat,750x,075,f-pad,750x1000,f8f8f8.jpg

This way they would look different at each angle but still nice.

**Comment by runsy** [comment]
I like the Minetest aproach.

**Comment by kromka-chleba** [comment]
Adding non-symmetrical plants would be nice.
I often design plants for Exile and symmetry does not always help.
Perhaps someone could add an option to define a non-symmetrical stalk for plantlike nodes.
